fintrack/src/context/AuthContext.js

```diff
@@ -0,0 +1,35 @@
+import { createContext, useState, useEffect } from 'react'
+import Cookies from 'js-cookie'
+
+export const AuthContext = createContext(null)
+
+export const AuthProvider = ({ children }) => {
+
+    const [authToken, setAuthToken] = useState(null)
+
+    // load token from cookie when component mounts
+    useEffect(() => {
+        const storedToken = Cookies.get('authToken')
+        if (storedToken) {
+            setAuthToken(storedToken)
+        }
+    }, [])
+
```

Comment:
If `Cookies.get` was slow performance-wise, you might want a `useMemo` here instead of `useEffect`, just so that react re-renders don't cause this to be re-ran over and over (the call to `Cookies.get`, specifically).

However, this is not a slow browser API, so I wouldn't bother, and would suggest just calling it straight-up.

So, either this is a call to `useEffect` that should be `useMemo`, or probably I'd have no hook here at all.

Lots of `useEffect` abuse all over, `useEffect` is for _synchronization_ with outside things, like the server, and you almost never need to use it, in general, and should avoid it, because unnecessary usages of it always make things more confusing than they need to be!